vaultlocker/tests/unit/test_vaultlocker.py

```diff
@@ -140,3 +142,102 @@ def test_get_vault_path(self, _socket):
         _socket.gethostname.return_value = 'myhost'
         self.assertEqual(shell._get_vault_path('my-UUID', self.config),
                          'vaultlocker-test/myhost/my-UUID')
+
+    @mock.patch.object(shell, 'systemd')
+    @mock.patch.object(shell, 'dmcrypt')
+    @mock.patch.object(shell, '_get_vault_path')
+    def test_encrypt_luks_failure(self, _get_vault_path, _dmcrypt, _systemd):
+        _get_vault_path.return_value = 'backend/host/uuid'
+        _dmcrypt.generate_key.return_value = 'testkey'
+        _dmcrypt.luks_format.side_effect = \
+            subprocess.CalledProcessError(returncode=-1,
+                                          cmd="echo Unit Test")
+
+        args = mock.MagicMock()
+        args.uuid = 'passed-UUID'
+        args.block_device = ['/dev/sdb']
+
+        client = mock.MagicMock()
+        client.read.return_value = {
+            'data': {
+                'dmcrypt_key': 'testkey'
+            }
+        }
+
+        self.assertRaises(
+            exceptions.LUKSFailure,
+            shell._encrypt_block_device,
+            args, client, self.config
+        )
+
+        client.delete.assert_called_once_with('backend/host/uuid')
+
+    @mock.patch.object(shell, 'systemd')
+    @mock.patch.object(shell, 'dmcrypt')
+    @mock.patch.object(shell, '_get_vault_path')
+    def test_vault_write_operation(self, _get_vault_path,
+                                   _dmcrypt, _systemd):
+        _get_vault_path.return_value = 'backend/host/uuid'
+        _dmcrypt.generate_key.return_value = 'testkey'
+
+        args = mock.MagicMock()
+        args.uuid = 'passed-UUID'
+        args.block_device = ['/dev/sdb']
+
+        client = mock.MagicMock()
+        client.read.return_value = {
+            'data': {
+                'dmcrypt_key': 'testkey'
+            }
+        }
+
+        self.assertIsNot(
```

Comment:
You could just do

```
        # This will fail if it raises anything, so simply calling something
        # is enough to asserting that it doesn't raise:
        shell._encrypt_block_device(
            args,
            client,
            self.config)
```

Same below in other tests

Reply:
Discussed offline: I'm also fine with leaving this unchanged.